File: api/app/models/field.rb
class Field < ApplicationRecord
  belongs_to :form

  enum :field_type, { input: 0, textarea: 1 }

  validates :name, presence: true
  validates :label, presence: true
  validates :order, presence: true, numericality: { only_integer: true }
  validates :field_type, presence: true, inclusion: { in: field_types.keys }
  validates :is_required, inclusion: { in: [ true, false ] }
  validates :order, presence: true
end

File: frontend/src/lib/types/field.ts
export enum FieldType {
  Input = "input",
  Textarea = "textarea",
}

export const FieldTypeLabels: Record<FieldType, string> = {
  [FieldType.Input]: "テキスト",
  [FieldType.Textarea]: "テキストエリア",
}

export interface FieldParams {
  name: string
  label: string
  field_type: FieldType
  is_required: boolean
}

export interface Field {
  id: string
  form_id: string
  name: string
  label: string
  field_type: FieldType
  is_required: boolean
  order: number
}
